### DISCUSSION TITLE
Add Support for XenForo REST API

### DISCUSSION THREAD

**Discussion by RyleaStark** [VersoBit/XF-Badges#31] [Ideas]
Some forum administrators may wish to issue badges from third party sources; Here is a template of API functions we _could_ offer.

***

### `badges:give`
**Required Variables**: `badge_id`, `user_id`
**Optional Variables**: `reason`

Give a badge to a user.

_**Super User Key Required**_
***

### `badges:take`
**Variables**: `badge_id`, `user_id`

Take a badge from a user.

_**Super User Key Required**_
***

### `badges:categories`

Get a list of badge categories with badge counts.
***

### `badges:list`
**Optional _Super User Key_ Variables**: `user_id`, `category_id`

Get a list of badges to a specific user, or a list of badges specific to a category. This endpoint only provides the badges of the user key requesting; Super User Keys can request a list from specific users.

**Comment by LuckyRiver** [comment]
Hi, 

Can't wait to see this implemented as I am currently working on achievement system to integrate to my poker game.
The listed REST endpoints seem right for my needs.
Question for the list on the category? eg. Poker category assuming it is created along with the corresponding badges.
If user_id is omitted, does it list all the badges availables a user can earn ? 
And if submitted along with category, it lists all the badges a user has acquired?
My goal is using the REST API to runtime give badge to player. The badge is determined by poker game metrics which unlocks an particular achievement. I plan to have about 50 achievements, initially displayed as locked, as player earns badges, they will be displayed as unlocked. 
Looking forward to test drive this out using Postman and the super user keys.
Meanwhile, I have started coding the achievement part on my end and definitely  integrate the API calls when they are ready.
Thanks!

**Reply by RyleaStark** [reply]
‘badges:category’ would just list the badges available in the system by category and return a count of how many badges to expect.

‘badges:list’ would provide you either a list of badges (from a single category), or a list of badges the user has received.

**Comment by LuckyRiver** [comment]
Hi,

Any news when the REST API will be available?

Thanks!
